src/Analysis/Engine/Impl/Parsing/Ast/DottedName.cs

```diff
@@ -22,16 +22,14 @@ namespace Microsoft.PythonTools.Parsing.Ast {
     public class DottedName : Node {
         private readonly NameExpression[] _names;
 
-        public DottedName(NameExpression[] names) {
+        public DottedName(NameExpression[]/*!*/ names) {
```

Comment:
What's the meaning of `/*!*/` introduced in a few of these files?

Reply:
Just a comment that I've used to mark parameter that can't be null. I can add `Check` calls instead.

Reply:
You don't have to change it, I just hadn't seen that before and was curious. Grepping through the project shows its use a number of other times as well (so I just hadn't noticed).

Reply:
To be clear - this is just a comment. It isn't handled by any analyzers. I really expect to see non-nullable reference types in C# 8: https://github.com/dotnet/csharplang/wiki/Nullable-Reference-Types-Preview

Reply:
Yep, I know. I actually went searching to see if that feature existed yet when you mentioned it. 🙂